### PR TITLE
Fix: Redeem manager migration mainnet - PE-50

### PR DESCRIPTION
**Pull request by koderholic**
## Description
Adapt bug fixes here to work for mainnet.

## Notice

- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/liquid-collective/liquid-collective-protocol/pulls) for the same update/change?
- [x] Have you assigned this PR to yourself?
- [x] Have you added at least 1 reviewer?
- [ ] Have you updated the official documentation?
- [ ] Have you added sufficient documentation in your code?
- [x] Have you added relevant tests to the official test suite?

## Pull Request Type

- [ ] 💫 New Feature (Breaking Change)
- [ ] 💫 New Feature (Non-breaking Change)
- [x] 🛠️ Bug fix (Non-breaking Change: Fixes an issue)
- [ ] 🕹️ Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)

## Testing

- [x] Have you tested this code with the official test suite?
- [ ] Have you tested this code manually?

### CODE REVIEW

File: contracts/src/RedeemManager.1.sol [hunk 1]
```diff
@@ -72,17 +71,14 @@ contract RedeemManagerV1 is Initializable, IRedeemManagerV1, IProtocolVersion {
 
     function _redeemQueueMigrationV1_2(address[] memory _prevInitiators) internal {
         RedeemQueueV1.RedeemRequest[] memory initialQueue = RedeemQueueV1.get();
```
Comment (iamsahu) [inline contracts/src/RedeemManager.1.sol:73]: I think we should rename it to `oldQueue` to make it more readable. It is easier to follow the code the when we have old & new vs initial & new.

File: contracts/src/RedeemManager.1.sol [hunk 2]
```diff
@@ -72,17 +71,14 @@ contract RedeemManagerV1 is Initializable, IRedeemManagerV1, IProtocolVersion {
...
-        RedeemQueueV1_2.RedeemRequest[] memory currentQueue = RedeemQueueV1_2.get(); //TODO: Remove after dev upgrade, not needed for staging/prod
-        uint256 currentQueueLen = currentQueue.length;
+        uint256 currentQueueLen = initialQueue.length;
```
Comment (iamsahu) [inline contracts/src/RedeemManager.1.sol:74]: Can we please rename this to `oldQueueLen`. Following the last suggestion.

File: contracts/test/fork/mainnet/4.redeemQueueMigrationV1_2_1.t.sol [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        return randomAddresses;
+    }
+
+    modifier shouldSkip() {
+        if (!_skip) {
+            _;
+        }
+    }
+
+    function test_migrate_allRequestInOneCall() external shouldSkip {
```
Comment (iamsahu) [inline contracts/test/fork/mainnet/4.redeemQueueMigrationV1_2_1.t.sol:60]: I think we should check all of the redeem requests rather than a few of them.

**Review by iamsahu — CHANGES_REQUESTED** [review]
It looks good. Only a few name changes & an expansion to a test.

File: contracts/src/RedeemManager.1.sol (hunk 1 above)
Comment (koderholic) [reply contracts/src/RedeemManager.1.sol:73]: Makes sense, would update.

File: contracts/test/fork/mainnet/4.redeemQueueMigrationV1_2_1.t.sol (hunk 3 above)
Comment (juliaaschmidt) [reply contracts/test/fork/mainnet/4.redeemQueueMigrationV1_2_1.t.sol:60]: maybe rename the test as well to make clearer what it's checking

**Review by iamsahu — DISMISSED** [review]
LGTM! 🚀

**Review by juliaaschmidt — DISMISSED** [review]
LGTM 🚀

**Review by iamsahu — APPROVED** [review]
LGTM

**Review by juliaaschmidt — APPROVED** [review]
🚀

**Pull request merged**